Add log statements to IndexedDbProvider open

balaji-viswanathan opened the pull request:
We see that there are timeouts and failures happen when clients call `indexeddbprovider.open`. We'll add logging at different points in db open method to understand where exactly the time is spent.

M1Les commented on src/IndexedDbProvider.ts:
```diff
@@ -155,20 +155,28 @@ export class IndexedDbProvider extends DbProvider {
     }
 
     if (wipeIfExists) {
+      this.logger.log(`Wiping db: ${dbName}`);
```
Somewhat, yeah. Every log statement is coming from either a db, object store, or an index. So we can set up context to have  dbName, if it is logged from db, dbName+storeName if it is coming from store, dbName+storeName+indexName, if it is coming from index. And then when we actually call the `this.logger.log` method, we will format string like so: (messages are just random, no connection to the actual ones, just an example)

"Wiping DB. DB:dbName"
"Error creating index indexName. DB:dbName, Store:storeName"
"Error removing value. DB:dbName, Store:storeName, Index:indexName"

So the message will have some dynamic parameters as well, but the suffix will have the context of the level where it was logged from.

balaji-viswanathan replied:
Implemented this using `LogWriter` class. Let me know if I have missed anything. Ready for review.

balaji-viswanathan commented:
@M1Les Can you review this now? Added `LogWriter` class which will take in the `context` and generate a message that can be written to logs. Added tests as well.
